### PR TITLE
[Feature] Add lunarvim latest release tag to dashboard

### PR DESCRIPTION
**Pull request by chaesngmin**
# Description
- Add LunarVim's latest version tag to dashboard
- Refactor some VimL to Lua

Displays version based on,
``` bash
git describe --tag
``` 

image

### CODE REVIEW

File: lua/core/dashboard.lua [hunk 1]
```diff
@@ -69,14 +69,22 @@ M.setup = function()
 
   vim.g.dashboard_session_directory = lvim.builtin.dashboard.session_directory
 
-  vim.cmd "let packages = len(globpath('~/.local/share/lunarvim/site/pack/packer/start', '*', 0, 1))"
+  local lv_site = "lunarvim.org"
+  local lv_path = "$HOME/.local/share/lunarvim"
```
Comment (Tastyep) [inline lua/core/dashboard.lua:73]: Should be replaced by LUNARVIM_RUNTIME_DIR once https://github.com/LunarVim/LunarVim/pull/1381 is merged

File: lua/core/dashboard.lua [hunk 2]
```diff
@@ -69,14 +69,22 @@ M.setup = function()
...
 
-  vim.api.nvim_exec(
-    [[
-    let g:dashboard_custom_footer = ['LunarVim loaded '..packages..' plugins  ']
-]],
-    false
-  )
+  local lv_ver_full = vim.fn.system("git -C " .. lv_path .. "/lvim describe --tag")
```
Comment (Tastyep) [inline lua/core/dashboard.lua:75]: For the time being it's fine, but I wonder if we should not have this in a separate file (possibly autogenerated)

File: lua/core/dashboard.lua [hunk 3]
```diff
@@ -69,14 +69,22 @@ M.setup = function()
...
+  local lv_ver = vim.fn.split(lv_ver_full, "-")[1]
+  local num_plugins_loaded = #vim.fn.globpath(lv_path .. "/site/pack/packer/start", "*", 0, 1)
+
+  -- vim.g.dashboard_custom_footer = { "   v" .. lv_ver, "", lv_site .. "     " }
```
Comment (Tastyep) [inline lua/core/dashboard.lua:79]: Remove comment

**Review by Tastyep — CHANGES_REQUESTED** [review]

File: lua/core/dashboard.lua (hunk 1 above)
Comment (chaesngmin) [reply lua/core/dashboard.lua:73]: Ty, will keep my eyes on it :)

File: lua/core/dashboard.lua (hunk 2 above)
Comment (chaesngmin) [reply lua/core/dashboard.lua:75]: Where do you think is proper place to have this?
Would you explain more about autogeneration?

Comment (Tastyep) [reply lua/core/dashboard.lua:75]: A workflow could generate a version file on commits on master, for example.

Comment (chaesngmin) [reply lua/core/dashboard.lua]: I added a bootstrap function that returns LunarVim version instead.
I may be wrong but thought file I/O on every startup could be more expensive.

**Review by Tastyep — APPROVED** [review]
Fine with me.

I'm looking forward to moving these globals to the lvim/ module

**Comment by abzcoding** [comment]
@chaesngmin can you update fix the merge conflicts so we can merged this 🥺

**Pull request merged**